float64 range
int64 time
